DigitalLearningSolutions.Data/Services/LogoService.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System.Data;
+    using System.Linq;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models;
+
+    public interface ILogoService
+    {
+        Logo GetLogo(int? centreId, int? customisationId);
+    }
+
+    public class LogoService : ILogoService
+    {
+        private readonly IDbConnection connection;
+
+        public LogoService(IDbConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public Logo GetLogo(int? centreId, int? customisationId)
+        {
+            using var multi = connection.QueryMultiple(
+                @"SELECT Centres.CentreName AS LogoName,
+                         Centres.CentreLogo AS LogoData,
+                         Centres.LogoMimeType
+                    FROM Centres
+                   WHERE Centres.CentreID = @centreId;
+
+                  SELECT Brands.BrandName AS LogoName, Brands.BrandLogo AS LogoData
+                    FROM Customisations
+                         INNER JOIN Applications
+                         ON Applications.ApplicationID = Customisations.ApplicationID
+
+                         INNER JOIN Brands
+                         ON Applications.BrandID = Brands.BrandID
+                   WHERE Customisations.CustomisationID = @customisationId AND CentreID = @centreId;",

```

Comment:
Why use two queries instead of just one? 🙂

Reply:
It felt like I was querying two discrete pieces of data (the Centre Logo, and the Brand Logo) so two queries felt natural. 

I could join the queries on the `centreId` and have a logo constructor that takes two sets of logo information and stores the first non-null one, if you think that would be better? 🙂

Reply:
Yes, I think I would prefer that 🙂

Reply:
Refactored to this in 60805a3 🙂